src/rulesets/Base/array.jl

```diff
@@ -103,3 +103,40 @@ function rrule(::typeof(fill), value::Any, dims::Int...)
     end
     return fill(value, dims), fill_pullback
 end
+
+#####
+##### `repeat`
+#####
+
+function rrule(::typeof(repeat), x::AbstractVector, m::Integer)
+    function repeat_pullback(Ȳ)
+        return (NO_FIELDS, dropdims(sum(reshape(Ȳ, length(x), :); dims=2); dims=2), DoesNotExist())
+    end
+    return repeat(x, m), repeat_pullback
+end
+
+function rrule(::typeof(repeat), x::AbstractVecOrMat, m::Integer, n::Integer=1)
```

Comment:
Since things like `repeat(rand(Int8,2,2,2),1,1,2,2)` are allowed, it might be nice to treat the general case? Untested, but my attempt is: 
```
function rrule(::typeof(repeat), x::AbstractArray, scales::Integer...)
    function repeat_pullback_1(dy)
        size2ndims = ntuple(d -> isodd(d) ? size(x,1+d÷2) : get(scales,d÷2,1), 2*ndims(dy))
        sumdy = sum(reshape(dy, size2ndims); dims = ntuple(d -> 2d, ndims(dy)))
        return (NO_FIELDS, reshape(sumdy, size(x)), map(_->DoesNotExist(), scales)...)
    end
    return repeat(x, scales...), repeat_pullback_1
end
```
What this won't handle is `repeat(1:2, 3,2,1,0)` but perhaps nobody will write that!